src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -668,23 +669,20 @@ else if (child.getType() != JavadocTokenTypes.LEADING_ASTERISK
      * @return violation string.
      */
     private static String getSummarySentence(DetailNode ast) {
-        boolean flag = true;
         final StringBuilder result = new StringBuilder(256);
         for (DetailNode child : ast.getChildren()) {
-            if (ALLOWED_TYPES.contains(child.getType())) {
+            if (child.getType() == JavadocTokenTypes.JAVADOC_TAG) {
+                break;
+            }
+            if (child.getType() != JavadocTokenTypes.EOF
```

Comment:
I am not following where `EOF` came from.

Reply:
`EOL` value is `-1`, it is not compatible with `BitSet`, so it requires special processing.

Reply:
Thanks, that explains it.